File: factory-method/c#/Animals/UnknownAnimal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactoryMethod
{
    public class UnknownAnimal : IAnimal
    {
        public ICookedMeat Cook()
        {
            var meat = new MysteryMeat();
            meat.Weight = 10;
            return meat;
        }
    }
}

File: factory-method/typescript/src/Animals/UnknownAnimal.ts
import {IAnimal} from "Interfaces/IAnimal";
import {ICookedMeat} from "Interfaces/ICookedMeat";
import {MysteryMeat} from "Meats/MysteryMeat";

export class UnknownAnimal implements IAnimal{
  public Cook(): ICookedMeat{
    let meat : MysteryMeat = new MysteryMeat();
    meat.Weight = 10;
    return meat;
  }
}